app/src/main/java/com/g11x/checklistapp/ChecklistItemActivity.java

```diff
@@ -18,35 +18,57 @@
 package com.g11x.checklistapp;
 
 import android.content.Intent;
-import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
-import android.view.Menu;
-import android.view.MenuInflater;
-import android.view.MenuItem;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
 import android.view.View;
 import android.widget.Button;
 import android.widget.TextView;
 import android.widget.ToggleButton;
 
 import com.g11x.checklistapp.data.ChecklistItem;
 import com.g11x.checklistapp.data.ChecklistManager;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
 
 public class ChecklistItemActivity extends AppCompatActivity {
 
   private ChecklistItem checklistItem;
   private ToggleButton isDone;
+  private DatabaseReference databaseRef;
 
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     setContentView(R.layout.activity_checklist_item);
 
-    int index = this.getIntent().getIntExtra("index", 0);
-    checklistItem = ChecklistManager.get(getApplicationContext()).getItems().get(index);
+    databaseRef = FirebaseDatabase.getInstance().getReferenceFromUrl(
+        this.getIntent().getStringExtra("databaseRefUrl"));
+
+    databaseRef.addListenerForSingleValueEvent(new ValueEventListener() {
```

Comment:
Changing `addListenerForSingleValueEvent` to `addValueEventListener` makes this activity update its information when it changes server-side, so that seems preferable. Any objection to switching?